### ISSUE TITLE
Ports defining DISTVERSIONPREFIX don't seem to be correctly supported

### ISSUE THREAD

**Issue by alexdupre** [freebsd/portscout#4]
A new version is detected even if it's the same one:

![screenshot_20181217_164135](https://user-images.githubusercontent.com/3694371/50097658-a6a8db00-021a-11e9-8570-2a54acb4a247.png)

**Comment by koobs** [comment]
Some issues are somewhat expected, as explained in [FreeBSD Ports r405250](https://svnweb.freebsd.org/changeset/ports/405250) in a past update

```
...
Additionally it turns out, portscout currently only uses/stores a
normalized version ('ver') in its database, in its general design
attempt to be a generic version comparison tool

In particular, portscout does not reference or store PORTVERSION
or DISTVERSION{FULL}, so we have nothing 'canonical' to compare
the responses from Github (tags) against.

This change special-cases Github in the FindNewestFile subroutine, which
was obtained via Portroach [2] (OpenBSD's portscout fork).

Extending this, we also now only match version-esque looking strings
from the tag, in an attempt to normalise, because they come in many
forms, including {foo-}X.Y.Z{-bar}, foo_X_Y_Z, among others.
```

We may be able to tweak certain cases

**Comment by koobs** [comment]
@alexdupre @yurivict If you have noticed other examples where prefixes/suffixed/other version formats have caused different classes of issues, please add them here.

**Comment by yurivict** [comment]
misc/dynomite: ```0.6.21-rc2``` vs. ```v0.6.21-rc2```

**Comment by koobs** [comment]
> 
> 
> misc/dynomite: `0.6.21-rc2` vs. `v0.6.21-rc2` - listed as having a new version.

Yep that's covered in the first class (i edited the original issue summary with a checkbox)

If there are any other categories id be keen to know what they are

**Comment by yurivict** [comment]
When it comes to ```DISTVERSIONPREFIX``` this is the only kind that I saw.

**Comment by danfe** [comment]
Apparently it's still unfixed as of 2022, e.g. for several of my ports I see this:
![screenshot №1](https://user-images.githubusercontent.com/1132291/161011049-43ec095b-f129-49dd-b115-b3685a3ae267.png)

Possibly related bug is that `skipv` keyword is being ignored:
![screenshot №2](https://user-images.githubusercontent.com/1132291/161011039-21b8622c-b239-4a6d-894c-96297eb9d06f.png)
The port has `PORTSCOUT=skipv:2.1.0` specified.

**Comment by elgithubo** [comment]
It seems that this issue affects my port, too:

![20221204_19h07m58s_grim](https://user-images.githubusercontent.com/15913459/205508243-78f4308c-a734-435f-be5f-67aa6b79671d.png)
![20221204_19h09m44s_grim](https://user-images.githubusercontent.com/15913459/205508247-a21404c5-1859-4f03-84ca-3362dc0fee3c.png)
![20221204_19h19m30s_grim](https://user-images.githubusercontent.com/15913459/205508361-5facb18e-a381-4f91-b5ab-ee341e4f16e7.png)


I see no possible simple workaround using the PORTSCOUT Makefile variable which does not disable update notifications at all.
So I guess one just needs to ignore portscout in such cases?

**Comment by danfe** [comment]
> I see no possible simple workaround using the `PORTSCOUT` variable

There's nothing to work around here.  It's a bug in portscout which must be hunt down and fixed.

> So I guess one just needs to ignore portscout in such cases?

I prefer to not, as I want to see if/when something would change in this regard.

**Comment by tatsuki-makino** [comment]
It seems that the following explanation meant that this behavior was meant.
> - Any regexes or limitations below apply to just the version, not the
>  whole distfile name.

At least USE_GITHUB ports seem to have been stripped down [here](https://github.com/freebsd/portscout/blob/f1a90b0883ec7c8f712d11ed824450fea801d2d1/portscout.pl#L1008) with something like blenderbim- other than numbers and dots.
It looks like we'll need to pick it up as follows and implement a new key to match it.
`($prematchver, $postmatchver) = ($`\``, $');`
